docs/api.rst

```diff
@@ -3,7 +3,7 @@ API Reference
 
 .. module:: jwt
 
-.. function:: encode(payload, key, algorithm="HS256", headers=None, json_encoder=None)
+.. function:: encode(payload, key, algorithm="HS256", headers=None, json_encoder=None, typ="JWT")
```

Comment:
@dajiaji how about instead of adding `typ` as a top level kwarg, we instead allow changing it through `headers`?

Reply:
Sorry for my late reply.

> we instead allow changing it through headers?

To my understanding, we can already change "typ" through "headers", but the problem here is that you can't leave "typ" out. To solve this,  you mean `headers={"typ":""}`  is  better than `typ=""` ?

From my point of view, since there are now several standards for `typ` other than `JWT` (e.g., `dpop+jwt` and `scevent+jwt`), it is not a bad idea to be able to change `typ` as a top-level argument like `alg`.

Anyway, if you think `headers={"typ":""}` is better, I can update my PR for that. No problem. Feel free to request me.

Reply:
Hi @jpadilla can you please provide your feedback on the previous comment. Currently this is a blocker for me, and I'm  waiting for this PR to be merged into master. Thank you.

Reply:
Hi, @jpadilla. I fixed it as you requested. The changes are much less. Could you check and merge it?

Reply:
@jpadilla I resolved a conflict on CHANGELOG.